### PR TITLE
#4 added userContext, header/layout/footer components

### PR DESCRIPTION
**Pull request by heydut**

### CODE REVIEW

File: src/components/Layout/Layout.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Footer from "../Footer/Footer";
+import Header from "../Header/Header";
+
+const Layout = ({ children }) => {
+  return (
+    <>
+      <Header />
```
Comment (mz106) [inline src/components/Layout/Layout.jsx]: Please wrap wrap layout header, children and footer and add dummy state to layout component

File: src/index.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
-const root = ReactDOM.createRoot(document.getElementById('root'));
+import React from "react";
+import { UserContextProvider } from "./context/userContext.jsx";
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
   <React.StrictMode>
-    <App />
+    <UserContextProvider>
```
Comment (mz106) [inline src/index.js]: Please remove UserContext wrapper and swap with layout after the above changes

**Review by mz106 — CHANGES_REQUESTED** [review]
Dut, my bad. I know that you did what was in the issue but I got it wrong. Please see changes below. Sorry.

**Comment by mz106** [comment]
Also Dut, remember to assign yourself to the pull requests - your GH profile will have an increased number of pull requests if you do that.

**Comment by heydut** [comment]
What you asked looks a little confusing for me and I'm not sure I understood it correctly, but I made some changes.

**Comment by mz106** [comment]
Hi Dut. Please look at the following link;

https://github.com/mz106/testing-react-sequelize/blob/main/library-test-fe/src/components/layout/Layout.js

In here, the Layout component returns the header, children and footer. These are wrapped by the UserContext.Proveder. Then Layout component then wraps the App component in index.js;

https://github.com/mz106/testing-react-sequelize/blob/main/library-test-fe/src/index.js

This is so that multiple contexts can be hidden in the Layout component and also allow access to all pages for the header and footer, as well as any context required globally.

File: src/index.js (hunk 2 above)
Comment (mz106) [reply src/index.js]: Please see my below comment

**Review by mz106 — APPROVED** [review]
Approved. Thanks Dut. Good to merge.

File: src/components/Layout/Layout.jsx (hunk 1 above)
Comment (mz106) [reply src/components/Layout/Layout.jsx]: Ok to go.

**Pull request merged**